Spigot-Server-Patches/0613-Spawner-API-Nerf-Mobs.patch

```diff
@@ -0,0 +1,67 @@
+@@ -25,6 +25,7 @@ public abstract class MobSpawnerAbstract {
+     public int requiredPlayerRange = 16;
+     public int spawnRange = 4;
+     private int tickDelay = 0; // Paper
++    public boolean nerfMobs = false; private static final String PAPER_NERF_MOBS = "Paper.NerfMobs"; // Paper
+ 
+     public MobSpawnerAbstract() {}
+ 
+@@ -159,7 +160,7 @@ public abstract class MobSpawnerAbstract {
+                                     ((EntityInsentient) entity).prepare(worldserver, world.getDamageScaler(entity.getChunkCoordinates()), EnumMobSpawn.SPAWNER, (GroupDataEntity) null, (NBTTagCompound) null);
+                                 }
+                                 // Spigot Start
+-                                if ( entityinsentient.world.spigotConfig.nerfSpawnerMobs )
++                                if ( entityinsentient.world.spigotConfig.nerfSpawnerMobs || this.nerfMobs ) // Paper - Nerf Mobs API
+                                 {
+                                     entityinsentient.aware = false;
+                                 }
+@@ -268,6 +269,11 @@ public abstract class MobSpawnerAbstract {
+             this.j = null;
+         }
+ 
++        // Paper start - api for nerfing mobs
++        if (nbttagcompound.hasKeyOfType(PAPER_NERF_MOBS, org.bukkit.craftbukkit.util.CraftMagicNumbers.NBT.TAG_BYTE)) {
++            this.nerfMobs = nbttagcompound.getBoolean(PAPER_NERF_MOBS);
++        }
++        // Paper end
+     }
+ 
+     public NBTTagCompound b(NBTTagCompound nbttagcompound) {
+@@ -289,6 +295,7 @@ public abstract class MobSpawnerAbstract {
+ 
+             nbttagcompound.setShort("MinSpawnDelay", (short) Math.min(Short.MAX_VALUE, this.minSpawnDelay));
+             nbttagcompound.setShort("MaxSpawnDelay", (short) Math.min(Short.MAX_VALUE, this.maxSpawnDelay));
++            nbttagcompound.setBoolean(PAPER_NERF_MOBS, this.nerfMobs);
+             // Paper end
+             nbttagcompound.setShort("SpawnCount", (short) this.spawnCount);
+             nbttagcompound.setShort("MaxNearbyEntities", (short) this.maxNearbyEntities);
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftCreatureSpawner.java b/src/main/java/org/bukkit/craftbukkit/block/CraftCreatureSpawner.java
+index e78e3804bad9b0b6004939cadf40465dec68fdba..df01b292f5a35e945fb685667158df03d6d75bed 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftCreatureSpawner.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftCreatureSpawner.java
+@@ -132,5 +132,15 @@ public class CraftCreatureSpawner extends CraftBlockEntityState<TileEntityMobSpa
+     public void resetTimer() {
+         this.getSnapshot().getSpawner().resetTimer();
+     }
++
++    @Override
++    public boolean isNerfingMobs() {
++        return this.getSnapshot().getSpawner().nerfMobs;
++    }
++
++    @Override
++    public void setNerfingMobs(boolean nerfingMobs) {
++        this.getSnapshot().getSpawner().nerfMobs = nerfingMobs;
```

Comment:
does this actually work? since you modifying a snapshot and not the actual block?
wouldnt you need to update the snapshot?
(or is this the method that doesnt clone?)

Reply:
erm... the other methods in CraftCreatureSpawner follow the same style, so unless they also don't work, this should be fine